schema/queries.gql

```diff
@@ -155,6 +155,45 @@ type Query {
     """
     dangling: Boolean = false
   ): CompositionConnection!
+
+  """
+  Get an `XRMResource` and its descendants which form a tree.
+  """
+  xrmResourceTree(
+    "The `ID` of an `XRMResource`"
+    id: ID!
+  ): XRMResourceTreeConnection!
+}
+
+"""
+A XRM Resource which is either a `CompositeResource`, `CompositeResourceClaim`
+or `ManagedResource`
+"""
+union XRMResource = CompositeResource | CompositeResourceClaim | ManagedResource
```

Comment:
Could this be a `ProviderConfig` too? (I think yes - it's not uncommon to compose them).

Also, naming nit, maybe `CrossplaneResource`? It feels a little more consistent with the existing `KubernetesResource` (as opposed to `KRMResource`).